src/services/completions.ts

```diff
@@ -338,7 +338,12 @@ namespace ts.Completions {
     ): CompletionEntry | undefined {
         let insertText: string | undefined;
         let replacementSpan: TextSpan | undefined;
+
         const insertQuestionDot = origin && originIsNullableMember(origin);
+        if (insertQuestionDot && preferences === false) {

```

Comment:
I'm guessing this was supposed to be `preferences.includeAutomaticOptionalChainCompletions === false`.

Reply:
Wow, comparability strikes - `UserPreferences` is all-optional and weak type checks don't get applied in comparisons.

Reply:
Out of curiosity, why is *any* object type considered comparable to a primitive type like `boolean`?  I don't think the result of any such comparison could ever be meaningful, since objects are compared by reference at runtime (i.e. it would always be false).

Reply:
@fatcerberus an object type doesn't imply an object value. `"hello world"` is a valid `{ toString(): string }`

Reply:
No, I get why it’s *assignable*.  I’m confused why it’s *comparible*, as the comparison is never meaningful.

Does assignability always imply comparability?

Reply:
Assignability should imply comparability - at least, that was always the intent. Ideally the hierarchy is

comparable to &geq; assignable to &geq; a subtype of &geq; identical to